app/assets/javascripts/analytics/track-select-change.js

```diff
@@ -0,0 +1,32 @@
+(function () {
+  "use strict";
+
+  $(document).change(trackEvent);
+
+  function trackEvent(event) {
+    var $target = $(event.target);
+
+    if ($target.is("select")) {
+      validateTrackingId($target);
+      sendEventToDataLayer($target);
+    }
+  }
+
+  function validateTrackingId($select) {
+    var trackingId = $select.data("tracking-id");
```

Comment:
Another way is to be explicit about what select components are we tracking. 
Which might be a better option.

Reply:
> Instead of a warning message. Why don't we raise a failing test?

I'd thought about this, but can't think of an easy way of doing it. The simplest way I can think of is to enable JavaScript for _all_ of our tests (or maybe just all of our filter tests?), and then raise an exception in the JavaScript. The tests would then fail because of the JavaScript exception.

> be explicit about what select components are we tracking

I'd also thought about doing this, but Ivan said it was better to send everything. It also only works if we remember to add a test for the given select element, by which time we'll have probably remembered to add the `data-tracking-id` attribute.